packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -0,0 +1,156 @@
+        id: 'ratepayBirthDate',
+        label: 'payment.ratepay.birth_date',
+        required: true,
+        fieldType: DynamicFormFieldType.DATE,
+    },
+    {
+        name: 'ratepayPhoneCountryCode',
+        custom: false,
+        id: 'ratepayPhoneCountryCode',
+        label: 'payment.ratepay.phone_country_code',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        type: 'string',
+        maxLength: 2,
+    },
+    {
+        name: 'ratepayPhoneNumber',
+        custom: false,
+        id: 'ratepayPhoneNumber',
+        label: 'payment.ratepay.phone_number',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        maxLength: 11,
+        min: 8,
+    }
+];
+
+const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
+    method,
+    checkoutService,
+    paymentForm,
+    onUnhandledError,
+    paymentForm: {
+        isSubmitted,
+        setFieldValue,
+        setValidationSchema,
+        setSubmitted,
+    },
+    language,
+}) => {
+    const fieldsValues = useRef<Partial<RatePayFieldValues>>({});
+    const initializePayment = async () => {
+        try {
+            await checkoutService.initializePayment({
+                gatewayId: method.gateway,
+                methodId: method.id,
+                paypalcommerceratepay: {
+                    container: '#checkout-payment-continue',
+                    legalTextContainer: 'legal-text-container',
+                    getFieldsValues: () => fieldsValues.current,
+                    onRenderButton: () => {
+                        paymentForm.hidePaymentSubmitButton(method, true);
+                    },
+                    onError: (error: Error) => {
+                        paymentForm.disableSubmit(method, true);
+                        onUnhandledError(error);
+                    },
+                },
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    const deinitializePayment = async () => {
+        try {
+            await checkoutService.deinitializePayment({
+                gatewayId: method.gateway,
+                methodId: method.id,
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    useEffect(() => {
+        void initializePayment();
+
+        return () => {
+            void deinitializePayment();
+        };
+    }, []);
+
+    const handleChange = useCallback(
+        (fieldId: string) => (value: string) => {
+            setFieldValue(fieldId, value);
+            updateFieldValues({
+                [fieldId]: value
+            });
+
+        },
+        [setFieldValue],
+    );
+
+    const updateFieldValues = (field: { [key: string]: string }) => {
+        fieldsValues.current = {...fieldsValues.current, ...field};
+    }
```

Comment:
If `getFieldsValues: () => getFormValues()` (approach from the message above) will work, then this method can be removed